### ISSUE TITLE
[no-unused-vars] report when runtime var is only used within a type (`type MyType = typeof varName`)

### ISSUE THREAD

**Issue by happycollision** [typescript-eslint/typescript-eslint#3325]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

```TS
const foo = "bar"

export type Baz = typeof foo
```

Run it locally via https://github.com/happycollision/eslint-no-unused-vars-typescript-repro

**Expected Result**
An eslint error that `foo` is unused. (Just like it would if we removed the type def line.)

**Actual Result**
No eslint error reported.

**Additional Info**
If you agree with my thoughts below, then this is a bug. If you disagree, then I guess it is a feature request that'd need some options to opt into it.

I am of the opinion that if you create a runtime variable, you ought to be expected to use that variable within your runtime code. Using a runtime variable for purely type definition purposes means that when your code is compiled (depending on your compiler of choice and options), you'll have an extra variable handing around that does nothing. In all likelihood, this is an error and you _meant_ to use the variable but forgot to add it to whatever other part of your code needed it. This is exactly why I realized that this bug exists. 😄 

I cannot think of a single case where you'd _want_ such behavior, but I am totally willing to believe they might exist.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.22.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.25.0` |
| `node`                             | `v11.12.0` |

**Comment by bradzacher** [comment]
I think this should report in the rule, but it should have a different message.
Something like "value variable foo is only used in a type position", so that it's clear WHY it's being reported on.

**Comment by auvred** [comment]
#9330 was merged into the `v8` branch. ✅
